mxcubeweb/core/server/resource_handler.py

```diff
@@ -511,3 +495,142 @@ def register_blueprint(self, parent_bp) -> None:
             f"Blueprint '{self._bp.name}' ({self._url_prefix}) registered with server."
         )
         log.debug(msg)
+
+
+class ComponentResourceHandler(ResourceHandler):
+    """
+    AdapterResourceHandler is a Flask resource handler that dynamically creates routes
+    for hardware objects based on their attributes and methods.
+    It supports GET and PUT requests for attributes and commands respectively.
+    """
+
+    def __init__(  # noqa: PLR0913
+        self,
+        name: str,
+        url_prefix: str,
+        handler_dict: dict[str, object],
+        app: object,
+        exports: list[dict[str, str]],
+        commands: list[str],
+        attributes: list[str],
+    ) -> None:
+        super().__init__(
+            name, url_prefix, handler_dict, app, exports, commands, attributes
+        )
+
+    def _get_handler_object_route(self, export) -> str:
+        """
+        Returns the base route for the resource handler.
+        """
+        return f"{export['url']}"
+
+    def _create_view_func(self, export: dict[str, str]) -> Callable:
```

Comment:
The creation of the view function is slightly different between the two resource handlers. The most important difference is that there is only one instance of a `Component`, while an Adapter handles several instance of `HardwareObjects`. The hardware object id string is hence not needed for the `Component`.